bgpd/bgp_bmp.c

```diff
@@ -2223,7 +2223,7 @@ static void bmp_bgp_peer_vrf(struct bmp_bgp_peer *bbpeer, struct bgp *bgp)
 		send_holdtime = peer->bgp->default_holdtime;
 
 	/* for loc-rib, AS used is always the one from the current BGP instance */
-	s = bgp_open_make(peer, send_holdtime, bgp->as, &peer->local_id);
+	s = bgp_open_make(peer, send_holdtime, bgp->as, &bgp->router_id);
```

Comment:
Again we should use the local_id and have it set appropriately.  Why is this not being updated instead?

Reply:
there are other places that use the peer->local_id.  This change *implies* that those are now wrong too when they are being used.  Let's fix that?

Reply:
I keep the usage of peer->local_id, and I will maintain bgp->peer_self->router_id.